avocado/core/runner.py

```diff
@@ -468,6 +469,12 @@ def _iter_variants(template, mux):
         :raises ValueError: When variant and template declare params.
         """
         for variant, params in mux.itertests():
+            variant_name = ''
+            for x in params[0]:
+                path = x.path[len('/run'):]
+                variant_name += path.replace('/', '-')
+            # trim first '-'
+            variant_name = variant_name[1:]
```

Comment:
Yep, you can see the card about this here: https://trello.com/c/1lzDCiqn/835-improve-the-variant-ids and the motivation here: https://trello.com/c/wiGkOFSa/828-test-s-data-directory-should-include-the-variant-id

I don't like the names much as they are not unique enough. I'd use hash of all variants including their environments and then I'd show partial hashes of those in the UI.

I know that leaf names would be more user friendly, so how about `$names-$hash`? The only issue I have with the leaf names is that with multiple variants they tend to grow in size and soon we reach the filesystem limit. As a solution how about using hashes internally including storing a file `variant` containing the hash in results, but for the UI we could show the leaf names?